src/SfxWeb/src/app/Models/DataModels/collections/infrastructureCollection.ts

```diff
@@ -0,0 +1,106 @@
+    return of(null)
+  }
+}
+
+export class InfrastructureCollection extends DataModelCollectionBase<InfrastructureCollectionItem> {
+  static readonly bannerThrottledJobs = 'throttled-banner';
+
+  public throttledJobs: InfrastructureJob[];
+
+  public constructor(data: DataService) {
+    super(data, parent);
+  }
+
+  public static stripPrefix(name: string) {
+    return name.replace(`fabric:/System/InfrastructureService/`, '')
+  }
+
+  protected retrieveNewCollection(messageHandler?: IResponseMessageHandler): Observable<any> {
+    return this.data.getSystemServices(true, messageHandler).pipe(mergeMap(services => {
+      const infrastructureServices = services.collection.filter(service => service.raw.TypeName === Constants.InfrastructureServiceType);
+      return forkJoin(infrastructureServices.map(service => this.data.restClient.getInfrastructureJobs(service.id).pipe
+        (map(items => {
+          return new InfrastructureCollectionItem(this.data, {
+            Name: service.name,
+            Jobs: items.map(item => new InfrastructureJob(this.data, item))
+          })
+        }))))
+    }))
+  }
+
+  protected updateInternal(): Observable<any> {
+    this.throttledJobs = [];
+
+    const throttledIS = this.collection.filter(infra => infra.isThrottled);
+    if (throttledIS.length > 0) {
+      throttledIS.forEach(is => this.throttledJobs = this.throttledJobs.concat(is.throttledJobs));
+
+      this.data.warnings.addOrUpdateNotification({
+        message: `Active updates count has exceeded the max allowed for safe rollout of updates for
+                    ${throttledIS.map(is => is.name).join(",")}
+                    Once the existing updates complete, the pending updates will start automatically `,
+        level: StatusWarningLevel.Warning,
+        priority: 4,
+        id: InfrastructureCollection.bannerThrottledJobs,
+      });
+    } else {
+      this.data.warnings.removeNotificationById(InfrastructureCollection.bannerThrottledJobs);
+    }
+    return this.data.getNodes().pipe(map(collection => {
+      const counter = new Counter();
+      collection.collection.forEach(node => {
+        counter.add(node.raw.Type);
+      })
+
+      const nodetypesWithoutEnoughNodes = this.collection.map(is => InfrastructureCollection.stripPrefix(is.name)).filter(is => (counter.entries().find(count => count.key === is)?.value || 0) < 5);
+
+      if (nodetypesWithoutEnoughNodes.length > 0) {
+        this.data.warnings.addOrUpdateNotification({
+          message: `Nodetype${nodetypesWithoutEnoughNodes.length > 1 ? 's' : ''} ${nodetypesWithoutEnoughNodes.join()} ${nodetypesWithoutEnoughNodes.length > 1 ? 'are' : 'is'} deployed with less than 5 VMs. Atleast 5 VMs in the VMSS are required to be present for the platforms updates to work reliably.
+                    Please fix this misconfiguration as updates to such VMSS will be blocked soon and deployments will start failing. `,
```

Comment:
For details: https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-capacity#durability-characteristics-of-the-cluster”.